### PR TITLE
T34555: Adding sites from gnome-pwa-lists

### PR DESCRIPTION
**Pull request by MichaelDrydenEndless**
Added sites from the gnome-pwa-lists to [eos-extra-pwa.yaml](https://github.com/endlessm/gnome-software-data/blob/master/app-info/eos-extra-pwa.yaml) and made the from_head functions a bit more robust and versatile

https://phabricator.endlessm.com/T34555

### CODE REVIEW

File: app-info/metainfo/org.gnome.Software.WebApp_e41134be0ce95da0832878faa90cf3473b1b52d7.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<?xml version='1.0' encoding='utf-8'?>
+<component type="web-application">
+  <id>org.gnome.Software.WebApp_e41134be0ce95da0832878faa90cf3473b1b52d7.desktop</id>
+  <name>Squoosh</name>
+  <launchable type="url">https://squoosh.app/</launchable>
+  <url type="homepage">https://squoosh.app/</url>
+  <summary>Squoosh</summary>
```
Comment (wjt) [inline app-info/metainfo/org.gnome.Software.WebApp_e41134be0ce95da0832878faa90cf3473b1b52d7.metainfo.xml:7]: Not a great summary :)

File: app-info/metainfo/org.gnome.Software.WebApp_e636aa5f2069f6e9c02deccc7b65f43da7985e32.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<?xml version='1.0' encoding='utf-8'?>
+<component type="web-application">
+  <id>org.gnome.Software.WebApp_e636aa5f2069f6e9c02deccc7b65f43da7985e32.desktop</id>
+  <name>Pinafore for Mastodon</name>
+  <launchable type="url">https://pinafore.social/</launchable>
+  <url type="homepage">https://pinafore.social/</url>
+  <summary>An alternative web client for Mastodon, focused on speed and simplicity</summary>
+  <description>An alternative web client for Mastodon, focused on speed and simplicity.</description>
```
Comment (wjt) [inline app-info/metainfo/org.gnome.Software.WebApp_e636aa5f2069f6e9c02deccc7b65f43da7985e32.metainfo.xml:8]: Summary and description should not be the same string

File: app-info/metainfo/org.gnome.Software.WebApp_e730d810bf154c7818b460479ea7826140099a7a.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<?xml version='1.0' encoding='utf-8'?>
+<component type="web-application">
+  <id>org.gnome.Software.WebApp_e730d810bf154c7818b460479ea7826140099a7a.desktop</id>
+  <name>Instagram</name>
+  <launchable type="url">https://www.instagram.com/</launchable>
+  <url type="homepage">https://www.instagram.com/</url>
+  <summary>Instagram</summary>
```
Comment (wjt) [inline app-info/metainfo/org.gnome.Software.WebApp_e730d810bf154c7818b460479ea7826140099a7a.metainfo.xml:7]: Summary should not be the same as the name

File: app-info/metainfo/org.gnome.Software.WebApp_e730d810bf154c7818b460479ea7826140099a7a.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  <description>Create an account or log in to Instagram - A simple, fun &amp; creative way to capture, edit &amp; share photos, videos &amp; messages with friends &amp; family.</description>
```
Comment (wjt) [inline app-info/metainfo/org.gnome.Software.WebApp_e730d810bf154c7818b460479ea7826140099a7a.metainfo.xml:8]: "create an account or log into Instagram" should be removed. The rest of the description looks fine, could also be the summary

File: app-info/metainfo/org.gnome.Software.WebApp_ea52ad18346faac27072cd1654243dfb7c70ef14.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,49 @@
+<?xml version='1.0' encoding='utf-8'?>
+<component type="web-application">
+  <id>org.gnome.Software.WebApp_ea52ad18346faac27072cd1654243dfb7c70ef14.desktop</id>
+  <name>Excalidraw</name>
+  <launchable type="url">https://excalidraw.com/</launchable>
+  <url type="homepage">https://excalidraw.com/</url>
+  <summary>Excalidraw — Collaborative whiteboarding made easy</summary>
```
Comment (wjt) [inline app-info/metainfo/org.gnome.Software.WebApp_ea52ad18346faac27072cd1654243dfb7c70ef14.metainfo.xml:7]:
```suggestion
  <summary>Collaborative whiteboarding made easy</summary>
```

File: app-info/pwa-metainfo-generator.py [hunk 6]
```diff
@@ -132,21 +132,42 @@ def get_manifest(soup, url):
...
+        keywords = soup.head.find("meta", attrs={"name": "keywords"})
+        if keywords is None:
+            keywords = []
+        elif "meta" in str(keywords):
+            keywords = keywords['content'].split(", ")
+        else:
+            keywords = []
+    elif "meta" in str(keywords):
+        keywords = keywords['content'].split(", ")
+    else:
```
Comment (dbnicholson) [inline app-info/pwa-metainfo-generator.py:170]: This whole block looks exactly like what `prop_from_head` is doing and it appears `prop_from_head` isn't used anywhere. I think this whole part could be written like:

```
keywords_value = prop_from_head("keywords")
if keywords_value:
    keywords = keywords_value.split(", ")
else:
    keywords = []
```

File: app-info/eos-extra-pwa.yaml [hunk 7]
```diff
@@ -79,7 +79,8 @@
   adaptive: true
 - url: https://pinafore.social/
   name_property: name
-  description: An alternative web client for Mastodon, focused on speed and simplicity.
+  summary: An alternative web client for Mastodon, focused on speed and simplicity.
+  description:
```
Comment (dbnicholson) [inline app-info/eos-extra-pwa.yaml:83]: I think you need to come up with some description here. You also need to run `pwa-metainfo-generator.py` again and commit the changes to the metainfo files.

**Review by dbnicholson — CHANGES_REQUESTED** [review]
Looking good, but I think a little more polish is needed.

**Review by wjt — CHANGES_REQUESTED** [review]
Some of the categories are not quite right. I left notes on a couple but that's not exhaustive, e.g. Instagram has the same problem with comma-separated names in a single `<category>` tag rather than one `<category>` tag per category. This is probably due to not formatting the input YAML quite right.

**Review by dbnicholson — APPROVED** [review]
I'm going to merge this as is since we have too many outstanding branches. I think once we get all the desired apps in, we can take another sweep through the appstream to really polish it.

**Pull request merged**